skbio/sequence/_nucleotide_mixin.py

```diff
@@ -230,6 +230,87 @@ def is_reverse_complement(self, other):
             # underlying sequence data
             return self.reverse_complement()._string == other._string
 
+    def gc_content(self):
+        """Calculate the relative frequency of G's and C's in the sequence.
+
+        This includes G, C, and S characters. This is equivalent to calling
+        ``gc_frequency(relative=True)``.
+
+        Returns
+        -------
+        float
+            Relative frequency of G's and C's in the sequence.
+
+        See Also
+        --------
+        gc_frequency
+
+        Examples
+        --------
+        >>> from skbio import DNA
+        >>> DNA('ACGT').gc_content()
+        0.5
+        >>> DNA('ACGTACGT').gc_content()
+        0.5
+        >>> DNA('ACTTAGTT').gc_content()
+        0.25
+        >>> DNA('ACGT--..').gc_content()
+        0.5
+        >>> DNA('--..').gc_content()
+        0
+
+        """
+        return self.gc_frequency(relative=True)
+
+    def gc_frequency(self, relative=False):
+        """Calculate frequency of G's and C's in the sequence.
+
+        This calculates the minimum GC frequency, which corresponds to IUPAC
+        characters G, C, and S (which stands for G or C).
+
+        Parameters
+        ----------
+        relative : bool, optional
+            If False return the frequency of G, C, and S characters (ie the
+            count). If True return the relative frequency, ie the proportion
+            of G, C, and S characters in the sequence. In this case the
+            sequence will also be degapped before the operation, so gap
+            characters will not be included when calculating the length of the
+            sequence.
+
+        Returns
+        -------
+        int or float
+            Either frequency (count) or relative frequency (proportion),
+            depending on `relative`.
+
+        See Also
+        --------
+        gc_content
+
+        Examples
+        --------
+        >>> from skbio import DNA
+        >>> DNA('ACGT').gc_frequency()
+        2
+        >>> DNA('ACGT').gc_frequency(relative=True)
+        0.5
+        >>> DNA('ACGT--..').gc_frequency(relative=True)
+        0.5
+        >>> DNA('--..').gc_frequency(relative=True)
+        0
+
+        """
+
+        counts = np.bincount(self._bytes,
+                             minlength=self._number_of_extended_ascii_codes)
+        gc_ord = (ord(y) for y in 'CGS')
```

Comment:
Can you cache this value as a `classproperty`? This would be similar to how `IUPACSequence` is handling gap chars, degenerates, etc. (example [here](https://github.com/biocore/scikit-bio/blob/master/skbio/sequence/_iupac_sequence.py#L92-L97)).